sbnci/Modules/TPCSim/TPCSimValidation_module.cc

```diff
@@ -0,0 +1,246 @@
+
+
+// Framework includes
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Core/EDAnalyzer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/View.h"
+#include "canvas/Persistency/Common/Ptr.h"
+#include "canvas/Persistency/Common/PtrVector.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art_root_io/TFileService.h"
+#include "art_root_io/TFileDirectory.h"
+#include "canvas/Persistency/Common/FindMany.h"
+#include "canvas/Persistency/Common/FindOneP.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+//Larsoft includes
+#include "larcore/Geometry/Geometry.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+#include "larsim/MCCheater/BackTrackerService.h"
+#include "larsim/MCCheater/ParticleInventoryService.h"
+#include "lardata/DetectorInfoServices/DetectorClocksService.h"
+#include "larcoreobj/SimpleTypesAndConstants/geo_types.h"
+#include "sbndcode/RecoUtils/RecoUtils.h"
+#include "lardataobj/RawData/RawDigit.h"
+
+//Root Includes
+#include "TMath.h"
+#include "TTree.h"
+#include "TSystem.h"
+#include "TH1F.h"
+
+//C++ Includes
+#include <vector>
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+
+namespace ana {
+  class TPCSimValidation;
+}
+
+class ana::TPCSimValidation : public art::EDAnalyzer {
+  public:
+
+    explicit TPCSimValidation(const fhicl::ParameterSet& pset);
+    // The compiler-generated destructor is fine for non-base
+    // classes without bare pointers or other resource use.
+
+    // Plugins should not be copied or assigned.
+    TPCSimValidation(TPCSimValidation const&) = delete;
+    TPCSimValidation(TPCSimValidation&&) = delete;
+    TPCSimValidation& operator=(TPCSimValidation const&) = delete;
+    TPCSimValidation& operator=(TPCSimValidation&&) = delete;
+
+    // required abstract methods we need to impliment 
+    void analyze(const art::Event& evt) override;
+    void endJob() override;
+    void beginJob() override;
+
+  private:
+
+    //fcl parameters
+    string fGenieGenModuleLabel;
+    string fLArGeantModuleLabel;
+    string fRawDigitLabel;
+    bool   fVerbose;
+
+    //TTree
+    TTree* fTree;
+
+    //Service handlesacktracker
+    art::ServiceHandle<cheat::BackTrackerService> backtracker;
+    art::ServiceHandle<cheat::ParticleInventoryService> particleInventory;
+    art::ServiceHandle<geo::Geometry> geom;
+    art::ServiceHandle<art::TFileService> tfs;
+
+    int numevents;
+
+    // tree variables
+    int fRun;    ///< art run number
+    int fSubRun; ///< art subrun number
+    int fEvent;  ///< art event number
+
+    int fNRawDigits; ///<Number of RawDigits in event
+//    vector<short> fADCs; ///<Reference to the compressed ADC count vector. not working right now.
```

Comment:
Are we keeping all this in as a TODO?

Reply:
yes. I image this is something the experts will want to address when they get to it.